### PR TITLE
Add Modelling elements to the UI

### PR DESCRIPTION
**Pull request by alvina-yang**

### CODE REVIEW

**Review by Ford-Robert — APPROVED** [review]
Excellent work! How will we pass these chosen features onto the higher level processes, will that be handled in another layer?

**Review by Shivesh777 — DISMISSED** [review]

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        </HBox.margin></Button>
+   <Text strokeType="OUTSIDE" strokeWidth="0.0" text="Epoch" translateX="25.0" translateY="5.0">
+      <font>
+         <Font size="18.0" />
+      </font>
+   </Text>
+   <Text fx:id="epochNumber" strokeType="OUTSIDE" strokeWidth="0.0" text="0" textAlignment="CENTER" translateX="50.0" translateY="8.0" wrappingWidth="73.095703125" />
+
+    <MenuBar prefHeight="28.0" prefWidth="651.0" translateX="-10.0">
+        <Menu text="Learning Rate">
```
Comment (Rishit-dagli) [inline src/main/resources/com/playground/playground/ml-parameters-view.fxml:38]: Instead of a menu maybe we can just make this a textbox, what do you think?

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            <MenuItem text="0.00001" />
```
Comment (Rishit-dagli) [inline src/main/resources/com/playground/playground/ml-parameters-view.fxml:39]: if we still keep it a menu I would strongly recommend using something like 1e-x

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            <MenuItem text="0.00001" />
+            <MenuItem text="0.0001" />
+            <MenuItem text="0.001" />
+            <MenuItem text="0.003" />
+            <MenuItem text="0.01" />
+            <MenuItem text="0.03" />
+            <MenuItem text="0.1" />
+            <MenuItem text="0.3" />
+            <MenuItem text="1" />
```
Comment (Rishit-dagli) [inline src/main/resources/com/playground/playground/ml-parameters-view.fxml:48]: A learning rate of 1 or more makes the program pretty obsolete at least for our application (I have actually never seen learning rates close to 1 or higher than 1 at all)

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            <MenuItem text="3" />
+            <MenuItem text="10" />
+        </Menu>
+        <Menu text="Activation">
+            <MenuItem text="ReLU" />
+            <MenuItem text="Tanh" />
+            <MenuItem text="Sigmoid" />
+            <MenuItem text="Linear" />
+        </Menu>
+        <Menu text="Regularization">
```
Comment (Rishit-dagli) [inline src/main/resources/com/playground/playground/ml-parameters-view.fxml:58]: What regularizer out of these do we mean by these?

- Kernel regularizer
- Bias regularizer
- Activity regularizer

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            <MenuItem text="None" />
+            <MenuItem text="L1" />
+            <MenuItem text="L2" />
+        </Menu>
+        <Menu text="Regularization Rate">
```
Comment (Rishit-dagli) [inline src/main/resources/com/playground/playground/ml-parameters-view.fxml:63]:
```suggestion
        <Menu text="Regularization Factor">
```

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            <MenuItem text="0.03" />
+            <MenuItem text="0.1" />
+            <MenuItem text="0.3" />
+            <MenuItem text="1" />
+            <MenuItem text="3" />
+            <MenuItem text="10" />
+        </Menu>
+        <Menu text="Problem Type">
+            <MenuItem text="Classification" />
+            <MenuItem text="Regression" />
```
Comment (Rishit-dagli) [inline src/main/resources/com/playground/playground/ml-parameters-view.fxml:77]: We won't be doing regression because we were thinking it would be too hard on the UI

```suggestion
```

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        <HBox.margin>
+            <Insets left="10.0" />
+        </HBox.margin></Button>
+   <Text strokeType="OUTSIDE" strokeWidth="0.0" text="Epoch" translateX="25.0" translateY="5.0">
+      <font>
+         <Font size="18.0" />
+      </font>
+   </Text>
+   <Text fx:id="epochNumber" strokeType="OUTSIDE" strokeWidth="0.0" text="0" textAlignment="CENTER" translateX="50.0" translateY="8.0" wrappingWidth="73.095703125" />
+
```
Comment (Rishit-dagli) [inline src/main/resources/com/playground/playground/ml-parameters-view.fxml:36]: Could we also do:

- batch size
- optimizer
- seed

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+<?import javafx.scene.layout.HBox?>
+<?import javafx.scene.text.Font?>
+<?import javafx.scene.text.Text?>
+
+<HBox prefHeight="68.0" prefWidth="1005.0" xmlns="http://javafx.com/javafx/20.0.1" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.playground.playground.MlParametersController">
+    <padding>
+        <Insets bottom="20.0" left="20.0" right="20.0" top="20.0" />
+    </padding>
+
+    <Button fx:id="rewindButton" styleClass="circle-button">
```
Comment (Rishit-dagli) [inline src/main/resources/com/playground/playground/ml-parameters-view.fxml:18]: Just wanted to say these look pretty nicely placed!

File: src/main/java/com/playground/playground/MlParametersController.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+     * The location used to resolve relative paths for the root object, or
+     * {@code null} if the location is not known.
+     *
+     * @param resources
+     * The resources used to localize the root object, or {@code null} if
+     * the root object was not localized.
+     */
+    public void initialize(URL location, ResourceBundle resources) {
+        // setting buttons
+        setButtonWithImage(rewindButton, "playground-images/rewind-button.png");
```
Comment (Rishit-dagli) [inline src/main/java/com/playground/playground/MlParametersController.java:36]: We shouldn't do PNGs at all, I would always recommend in decreasing order of recommendation:

1. Use icons from some icons library or native icons (almost always or always do this), if you have a new icon (which we don't) import SVGs to these icons
2. Use .svg directly or .webp
3. Use some other vector graphics forms like .eps
4. .apng or .avif but only for lossless animation sequences
5. .png
6. .jpeg (please never do this)

**Review by Rishit-dagli — CHANGES_REQUESTED** [review]

**Review by Rishit-dagli — COMMENTED** [review]
/area UI
/retitle Add Modelling elements to the UI
/cc snkrs23 could you also take a look a this
/hold until we fix the merge conflicts, @alvina-yang do you want me to help with this?

File: .idea/workspace.xml [hunk 10]
```diff
@@ -4,77 +4,259 @@
     <option name="autoReloadType" value="SELECTIVE" />
   </component>
   <component name="ChangeListManager">
-    <list default="true" id="243afb4c-938a-462c-adb2-9d3f4bb6b741" name="Changes" comment="">
-      <change afterPath="$PROJECT_DIR$/gradlew" afterDir="false" />
-      <change afterPath="$PROJECT_DIR$/src/main/java/HelloFX.java" afterDir="false" />
-      <change beforePath="$PROJECT_DIR$/build.gradle" beforeDir="false" afterPath="$PROJECT_DIR$/build.gradle" afterDir="false" />
-      <change beforePath="$PROJECT_DIR$/src/main/java/tutorial/HelloWorld.java" beforeDir="false" />
+    <list default="true" id="be2dd017-553b-4235-9fbd-87199c74970d" name="Changes" comment="Javadoc created for MainApplication.java.">
```
Comment (Rishit-dagli) [inline .idea/workspace.xml:7]: This file shouldn't be committed at all

**Comment by Rishit-dagli** [comment]
/area UI
/retitle Add Modelling elements to the UI
/cc snkrs23 could you also take a look a this
/hold until we fix the merge conflicts, @alvina-yang do you want me to help with this?

**Comment by Rishit-dagli** [comment]
/area UI
/retitle Add Modelling elements to the UI
/cc snkrs23 could you also take a look a this
/hold until we fix the merge conflicts, @alvina-yang do you want me to help with this?

**Comment by Rishit-dagli** [comment]
/retitle Add Modelling elements to the UI

**Comment by Rishit-dagli** [comment]
/retitle Add Modelling elements to the UI

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml (hunk 1 above)
Comment (snkrs23) [reply src/main/resources/com/playground/playground/ml-parameters-view.fxml:38]: I think it might be easier if we had a dropdown menu so people who aren't necessarily very familiar with Tensorflow could play around with the different options.

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml (hunk 2 above)
Comment (snkrs23) [reply src/main/resources/com/playground/playground/ml-parameters-view.fxml:39]: This would be difficult for users unfamiliar with exponential notation to use.

File: src/main/resources/com/playground/playground/ml-parameters-view.fxml (hunk 3 above)
Comment (snkrs23) [reply src/main/resources/com/playground/playground/ml-parameters-view.fxml:48]: Do you want to keep learning rates above 1? I'll keep them on the menu just in case.
